packages/headless/src/api/analytics/analytics.ts

```diff
@@ -199,7 +199,7 @@ export const getPageID = () => {
   const lastPageView = actions.reverse().find((action) => {
     return action.name === 'PageView';
   });
-  if (!lastPageView) {
+  if (!lastPageView?.value) {
     return '';
   }
```

Comment:
In what context is an empty pageview sent? If an invalid pageview is sent, do we prefer not to include a pageId with requests at all, or do we prefer to send the last valid pageId?

If it's the latter, then I would suggest this:
```suggestion
  const lastPageView = actions.reverse().find((action) => {
    return action.name === 'PageView' && action.value;
  });
  if (!lastPageView) {
    return '';
  }
```